Sources/ScipioKit/FileSystem.swift

```diff
@@ -0,0 +1,201 @@
+
+    /// Determines if the given URL is an existing regular file.
+    /// - Parameter path: The URL to check.
+    func isFile(_ path: URL) -> Bool
+
+    /// Determines if the given URL is a symbolic link.
+    /// - Parameter path: The URL to check.
+    func isSymlink(_ path: URL) -> Bool
+
+    /// Copies an item from one URL to another.
+    /// - Parameters:
+    ///   - fromURL: The source URL.
+    ///   - toURL: The destination URL.
+    func copy(from fromURL: URL, to toURL: URL) throws
+
+    /// Creates a directory at the specified URL.
+    /// - Parameters:
+    ///   - directoryPath: The directory URL to create.
+    ///   - recursive: If true, create intermediate directories as needed.
+    func createDirectory(_ directoryPath: URL, recursive: Bool) throws
+
+    /// Moves an item from one URL to another.
+    /// - Parameters:
+    ///   - fromURL: The source URL.
+    ///   - toURL: The destination URL.
+    func move(from fromURL: URL, to toURL: URL) throws
+
+    /// Returns the names of items in the specified directory.
+    /// - Parameter directoryPath: The directory URL to list.
+    /// - Returns: An array of file and directory names; order is undefined.
+    func getDirectoryContents(_ directoryPath: URL) throws -> [String]
+
+    /// Recursively removes the file or directory at the specified URL.
+    /// No error is thrown if the item does not exist.
+    /// - Parameter path: The URL of the item to remove.
+    func removeFileTree(_ path: URL) throws
+}
+
+extension FileSystem {
+    /// Creates a directory at the specified URL without creating intermediate directories.
+    /// - Parameter directoryPath: The directory URL to create.
+    /// - Throws: An error if directory creation fails.
+    func createDirectory(_ directoryPath: URL) throws {
+        try createDirectory(directoryPath, recursive: false)
+    }
+
+    /// Writes a UTF-8 encoded string to the file at the specified URL.
+    /// - Parameters:
+    ///   - path: The file URL to write to.
+    ///   - string: The string content to write.
+    /// - Throws: An error if the write operation fails.
+    func writeFileContents(_ path: URL, string: String) throws {
+        guard let data = string.data(using: .utf8) else {
+            throw FileSystemError.utf8EncodingFailed
+        }
+        try writeFileContents(path, data: data)
+    }
+
+    /// Checks if the item at the specified URL exists, following symbolic links.
+    /// - Parameter path: The URL to check.
+    /// - Returns: True if the path exists; false otherwise.
+    func exists(_ path: URL) -> Bool {
+        exists(path, followSymlink: true)
+    }
+}
+
+public struct LocalFileSystem: FileSystem {
+    public static let `default` = LocalFileSystem()
+
+    public var tempDirectory: URL {
+        URL(filePath: NSTemporaryDirectory())
+    }
+
+    public var cachesDirectory: URL? {
+        FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first
```

Comment:
Could you extract `FileManager.default` to the instance value?

Reply:
FileManager is not Sendable, so we can’t retain a FileManager instance as a property unless we use `@unchecked Sendable`.
LocalFileSystem in `swift-tools-support-core` also uses `FileManager.default` directly in each function.

https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/FileSystem.swift#L371